cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/ticket/accesstoken/AccessTokenImpl.java

```diff
@@ -0,0 +1,48 @@
+package org.jasig.cas.support.oauth.ticket.accesstoken;
+
+import org.jasig.cas.authentication.Authentication;
+import org.jasig.cas.authentication.principal.Service;
+import org.jasig.cas.ticket.ExpirationPolicy;
+import org.jasig.cas.support.oauth.ticket.code.OAuthCodeImpl;
+
+import javax.persistence.DiscriminatorColumn;
+import javax.persistence.DiscriminatorValue;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+
+/**
+ * An OAuth access token implementation.
+ *
+ * @author Jerome Leleu
+ * @since 4.3.0
+ */
+@Entity
+@Table(name="OAUTH_TOKENS")
+@DiscriminatorColumn(name="TYPE")
```

Comment:
Do you mean: `@Entity` and `@Table`? For access tokens, the discriminator value is different so it's mandatory, isn't it?